operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ControllerConfigurationOverrider.java

```diff
@@ -22,7 +22,7 @@
   private ResourceEventFilter<R> customResourcePredicate;
   private final ControllerConfiguration<R> original;
   private Duration reconciliationMaxInterval;
-  private final Map<String, DependentResourceSpec<?, ?>> dependentResourceSpecs;
+  private final LinkedHashMap<String, DependentResourceSpec<?, ?>> namedDependentResourceSpecs;
```

Comment:
Why use a map why not a list? as in ControllerConfiguration or DefaultControllerConfiguration: `List<DependentResourceSpec<?, ?>>`
DependentResourceSpec already contains the name. It's basically data duplication.

Would be also consistent with those 2 classes.

Reply:
Makes it easier to check if the spec is already present, find it again, naturally ensures that there is no duplication.